Create Dockerfiles for x86 and arm32v6 (RaspberryPi)

FoxRomeo opened the pull request:
Create Dockerfiles for x86 and arm32v6 (RaspberryPi)

FoxRomeo committed:
Create Dockerfile for RaspberryPi/arm32v6

Create Dockerfile for RaspberryPi/arm32v6

FoxRomeo committed:
Created crontab example

FoxRomeo committed:
Dockerfile for x86-based systems

FoxRomeo committed:
updated crontab example

dschmidt commented on Docker/Dockerfile.rpi:
```diff
@@ -0,0 +1,32 @@
+      git \
+      cron && \
+    cd /opt && \
+    git clone https://github.com/FoxRomeo/carddav2fb.git && \
```
If you want these Dockerfiles upstream, better not use your Fork in it ;-)

dschmidt commented on Docker/Dockerfile.rpi:
```diff
@@ -0,0 +1,32 @@
+    useradd --no-log-init -r -g carddav2fb carddav2fb && \
+    mkdir /opt/carddav2fb/share && \
+    chown carddav2fb:carddav2fb /opt/carddav2fb/share && \
+    rm -rf /var/lib/apt/lists/*
```
If git is only used for the checkout it could be removed afterwards to keep the image smaller. Also another option could be to dowbload a tarball from github instead.

dschmidt reviewed (commented):
From a quick look the only difference between the two docker files is the base image, right?

Maybe use an ARG instead... 
https://www.jeffgeerling.com/blog/2017/use-arg-dockerfile-dynamic-image-specification

I'm in no way associated with this project, I just stumbled accross this repo and your PR because I would have written a Dockerfile myself..

FoxRomeo committed:
Update Dockerfile.x86

FoxRomeo committed:
Update Dockerfile.rpi

FoxRomeo committed:
RPI change of base image and change from php7.0 to php7.2

FoxRomeo committed:
fixed typo

FoxRomeo committed:
switch from stretch to buster

FoxRomeo committed:
added root-login lock; just to be sure

FoxRomeo committed:
changed rpi Dockerfile to alpine, to stay arm23v6 compatible (aka RPi A/B/B+ and later)

FoxRomeo committed:
fix mkdir

FoxRomeo committed:
Update README.md

FoxRomeo committed:
Merge pull request #2 from carlos22/master

Update merge form fork base

FoxRomeo committed:
added Jenkinsfile and build.sh incl. push to additional repository

FoxRomeo committed:
fixed typo in build.sh

FoxRomeo committed:
fixed typo in build.sh

FoxRomeo committed:
fixed typo in build.sh

FoxRomeo committed:
added BuildStatus

FoxRomeo committed:
updated to latest helper script version

FoxRomeo committed:
updated timer

FoxRomeo committed:
added fallback_type as not all FritzBoxes have a type 'other'

FoxRomeo committed:
fallback_type & changed type order (#3)

* adding fallback_topic

* changed order of types, otherwise fax would be marked as home or work

* prepare merge with master

FoxRomeo committed:
updated readme

FoxRomeo committed:
updated to build docker images with Debian Buster and PHP7.3 (#4)

* adding fallback_topic

* switched back to stretch for some tests

* switched back to stretch for some tests

* changed order of types, otherwise fax would be marked as home or work

* prepare merge with master

* updated to latest Debian:Buster, and checked/updated to PHP7.3 on Debian and Alpine

FoxRomeo committed:
added * and # to the allowed characters in a phone number

FoxRomeo committed:
updated version and minimum required php (7.0 as stretch is still oldstable)

FoxRomeo committed:
added private repo

FoxRomeo committed:
workaround for 'new' requirements by alpine >=3.13

FoxRomeo committed:
changed build process

FoxRomeo committed:
updated structure

Commit:
PREPARING CHANGE RUNUSER FROM USER root TO cron

PREPARING CHANGE RUNUSER FROM USER root TO cron

Commit:
PREPARING CHANGE RUNUSER FROM USER root TO cron

PREPARING CHANGE RUNUSER FROM USER root TO cron

FoxRomeo committed:
preparing switch to non-root user

FoxRomeo committed:
Merge branch 'master' of https://github.com/FoxRomeo/carddav2fb

FoxRomeo committed:
preparing switch to non-root user

FoxRomeo committed:
fixed broken/missing xml dependencies

FoxRomeo committed:
fixed broken/missing xml dependencies

FoxRomeo committed:
fixed broken runuser

FoxRomeo committed:
fixed broken/missing ftp dependencies

FoxRomeo committed:
updated Dokerfile to get php version to use via environment variable (latest use php8 now)

FoxRomeo committed:
updated Dokerfile to get php version to use via environment variable (latest use php8 now)

Commit:
Fixed default usb_disk path and import of numbers with * (#1)

changed FRITZ/fonpix/ to /FRITZ/fonpix/ (otherweise empty/default usbdisk path will not work)

modified phone number import filter to allow * and # as part of phone numbers (for example used for internal numbers "**610" in FritzBox)
